rust/src/storage/file/node_file_storage/mod.rs

```diff
@@ -717,14 +779,46 @@ mod tests {
     }
 
     #[test]
-    fn open_recovers_closed_state() {
+    fn close_is_no_op_in_read_only_mode() {
```

Comment:
What is this really testing? Close is already a no-op if no `reserve` / `set` / `delete` was called. To ensure that it doesn't touch the files at all, it might be better to get the timestamps from the file and check that they don't change.